<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <author>Jianwei Guo</author>
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>974</width>
    <height>825</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="minimumSize">
   <size>
    <width>512</width>
    <height>512</height>
   </size>
  </property>
  <property name="maximumSize">
   <size>
    <width>2048</width>
    <height>2048</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>IPML2D</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout">
    <item row="0" column="0">
     <widget class="GlViewer" name="viewer" native="true">
      <property name="enabled">
       <bool>true</bool>
      </property>
      <property name="sizePolicy">
       <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="minimumSize">
       <size>
        <width>512</width>
        <height>512</height>
       </size>
      </property>
      <property name="maximumSize">
       <size>
        <width>512</width>
        <height>512</height>
       </size>
      </property>
      <property name="focusPolicy">
       <enum>Qt::StrongFocus</enum>
      </property>
      <property name="autoFillBackground">
       <bool>false</bool>
      </property>
      <property name="locale">
       <locale language="English" country="UnitedStates"/>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>974</width>
     <height>26</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
   </widget>
   <widget class="QMenu" name="menuView">
    <property name="title">
     <string>View</string>
    </property>
    <addaction name="actionOptionDialog"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuView"/>
  </widget>
  <widget class="QDockWidget" name="dockWidget_optionsDialog">
   <property name="minimumSize">
    <size>
     <width>150</width>
     <height>400</height>
    </size>
   </property>
   <property name="windowTitle">
    <string>Options</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>2</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents">
    <widget class="QGroupBox" name="groupBox_operation">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>170</y>
       <width>141</width>
       <height>131</height>
      </rect>
     </property>
     <property name="title">
      <string>GenerateData</string>
     </property>
     <widget class="QPushButton" name="pushButton_loadLSystem">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>10</y>
        <width>111</width>
        <height>23</height>
       </rect>
      </property>
      <property name="text">
       <string>Load L-system</string>
      </property>
     </widget>
     <widget class="QPushButton" name="pushButton_render">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>100</y>
        <width>111</width>
        <height>21</height>
       </rect>
      </property>
      <property name="text">
       <string>Render</string>
      </property>
     </widget>
     <widget class="QPushButton" name="pushButton_drawLsystem">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>40</y>
        <width>111</width>
        <height>21</height>
       </rect>
      </property>
      <property name="text">
       <string>Draw L-system</string>
      </property>
     </widget>
     <widget class="QPushButton" name="pushButton_regularize">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>70</y>
        <width>111</width>
        <height>23</height>
       </rect>
      </property>
      <property name="text">
       <string>Regularize</string>
      </property>
     </widget>
    </widget>
    <widget class="QGroupBox" name="groupBox_parameters">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>10</y>
       <width>141</width>
       <height>151</height>
      </rect>
     </property>
     <property name="title">
      <string>Parameters</string>
     </property>
     <widget class="QLabel" name="label_branchAngle">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>30</y>
        <width>91</width>
        <height>21</height>
       </rect>
      </property>
      <property name="text">
       <string>BranchAgnle:</string>
      </property>
     </widget>
     <widget class="QDoubleSpinBox" name="doubleSpinBox_branchAngle">
      <property name="geometry">
       <rect>
        <x>80</x>
        <y>30</y>
        <width>51</width>
        <height>20</height>
       </rect>
      </property>
      <property name="decimals">
       <number>2</number>
      </property>
      <property name="maximum">
       <double>90.000000000000000</double>
      </property>
      <property name="singleStep">
       <double>0.010000000000000</double>
      </property>
      <property name="value">
       <double>30.000000000000000</double>
      </property>
     </widget>
     <widget class="QSpinBox" name="spinBox_Iter">
      <property name="geometry">
       <rect>
        <x>70</x>
        <y>10</y>
        <width>51</width>
        <height>16</height>
       </rect>
      </property>
      <property name="maximum">
       <number>100</number>
      </property>
      <property name="value">
       <number>2</number>
      </property>
     </widget>
     <widget class="QLabel" name="label_iteration">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>10</y>
        <width>61</width>
        <height>21</height>
       </rect>
      </property>
      <property name="text">
       <string>Iteration:</string>
      </property>
     </widget>
     <widget class="QCheckBox" name="checkBox_useSetBranchAngle">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>110</y>
        <width>131</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Use default angle</string>
      </property>
      <property name="checked">
       <bool>true</bool>
      </property>
     </widget>
     <widget class="QDoubleSpinBox" name="doubleSpinBox_anglePerturbation">
      <property name="geometry">
       <rect>
        <x>80</x>
        <y>70</y>
        <width>51</width>
        <height>16</height>
       </rect>
      </property>
      <property name="maximum">
       <double>90.000000000000000</double>
      </property>
      <property name="singleStep">
       <double>0.010000000000000</double>
      </property>
     </widget>
     <widget class="QLabel" name="label_anglePerturbation">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>70</y>
        <width>91</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>AnglePertur:</string>
      </property>
     </widget>
     <widget class="QCheckBox" name="checkBox_useSetScale">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>130</y>
        <width>131</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Use default scale</string>
      </property>
      <property name="checked">
       <bool>true</bool>
      </property>
     </widget>
     <widget class="QDoubleSpinBox" name="doubleSpinBox_scale">
      <property name="geometry">
       <rect>
        <x>80</x>
        <y>50</y>
        <width>51</width>
        <height>16</height>
       </rect>
      </property>
      <property name="singleStep">
       <double>0.010000000000000</double>
      </property>
      <property name="value">
       <double>1.000000000000000</double>
      </property>
     </widget>
     <widget class="QLabel" name="label_scale">
      <property name="geometry">
       <rect>
        <x>40</x>
        <y>50</y>
        <width>41</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Scalar:</string>
      </property>
     </widget>
     <widget class="QDoubleSpinBox" name="doubleSpinBox">
      <property name="geometry">
       <rect>
        <x>80</x>
        <y>90</y>
        <width>51</width>
        <height>16</height>
       </rect>
      </property>
     </widget>
     <widget class="QLabel" name="label">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>90</y>
        <width>71</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>ScalePertur:</string>
      </property>
     </widget>
    </widget>
    <widget class="QGroupBox" name="groupBox_views">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>590</y>
       <width>161</width>
       <height>141</height>
      </rect>
     </property>
     <property name="title">
      <string>Views</string>
     </property>
     <widget class="QCheckBox" name="checkBox_showTreeRender">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>60</y>
        <width>121</width>
        <height>21</height>
       </rect>
      </property>
      <property name="text">
       <string> Show TreeRender</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_LineWidth">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>20</y>
        <width>71</width>
        <height>21</height>
       </rect>
      </property>
      <property name="text">
       <string>Line Width:</string>
      </property>
     </widget>
     <widget class="QDoubleSpinBox" name="doubleSpinBox_LineWidth">
      <property name="geometry">
       <rect>
        <x>80</x>
        <y>20</y>
        <width>51</width>
        <height>22</height>
       </rect>
      </property>
      <property name="decimals">
       <number>2</number>
      </property>
      <property name="minimum">
       <double>0.010000000000000</double>
      </property>
      <property name="maximum">
       <double>20.000000000000000</double>
      </property>
      <property name="singleStep">
       <double>1.000000000000000</double>
      </property>
      <property name="value">
       <double>2.000000000000000</double>
      </property>
     </widget>
     <widget class="QCheckBox" name="checkBox_showTree">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>40</y>
        <width>111</width>
        <height>21</height>
       </rect>
      </property>
      <property name="text">
       <string> Show Tree</string>
      </property>
      <property name="checked">
       <bool>true</bool>
      </property>
     </widget>
     <widget class="QCheckBox" name="checkBox_showImage">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>80</y>
        <width>111</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string> Show_Image</string>
      </property>
     </widget>
     <widget class="QCheckBox" name="checkBox_randomRule">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>100</y>
        <width>111</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string> Random_rules</string>
      </property>
      <property name="checked">
       <bool>false</bool>
      </property>
     </widget>
     <widget class="QCheckBox" name="checkBox_showSampledPoints">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>120</y>
        <width>131</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Show sampled points</string>
      </property>
     </widget>
    </widget>
    <widget class="QGroupBox" name="groupBox_parsing">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>310</y>
       <width>141</width>
       <height>271</height>
      </rect>
     </property>
     <property name="title">
      <string>Parsing</string>
     </property>
     <widget class="QPushButton" name="pushButton_loadParsing">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>20</y>
        <width>111</width>
        <height>21</height>
       </rect>
      </property>
      <property name="text">
       <string>Load image</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_bbxID">
      <property name="geometry">
       <rect>
        <x>30</x>
        <y>140</y>
        <width>41</width>
        <height>21</height>
       </rect>
      </property>
      <property name="text">
       <string>bbx ID:</string>
      </property>
     </widget>
     <widget class="QSpinBox" name="spinBox_bbxID">
      <property name="geometry">
       <rect>
        <x>80</x>
        <y>140</y>
        <width>51</width>
        <height>16</height>
       </rect>
      </property>
      <property name="minimum">
       <number>-1</number>
      </property>
      <property name="maximum">
       <number>200</number>
      </property>
      <property name="singleStep">
       <number>10</number>
      </property>
      <property name="value">
       <number>-1</number>
      </property>
      <property name="displayIntegerBase">
       <number>10</number>
      </property>
     </widget>
     <widget class="QPushButton" name="pushButton_parseGrammar">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>110</y>
        <width>111</width>
        <height>21</height>
       </rect>
      </property>
      <property name="text">
       <string>Parse</string>
      </property>
     </widget>
     <widget class="QPushButton" name="pushButton_buildGraph">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>50</y>
        <width>111</width>
        <height>23</height>
       </rect>
      </property>
      <property name="text">
       <string>Build graph</string>
      </property>
     </widget>
     <widget class="QDoubleSpinBox" name="doubleSpinBox_weightThrs">
      <property name="geometry">
       <rect>
        <x>80</x>
        <y>160</y>
        <width>51</width>
        <height>16</height>
       </rect>
      </property>
      <property name="maximum">
       <double>20.000000000000000</double>
      </property>
      <property name="singleStep">
       <double>0.010000000000000</double>
      </property>
      <property name="value">
       <double>1.000000000000000</double>
      </property>
     </widget>
     <widget class="QLabel" name="label_weightThrs">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>160</y>
        <width>71</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>weightThrs:</string>
      </property>
     </widget>
     <widget class="QDoubleSpinBox" name="doubleSpinBox_centerDisThrs">
      <property name="geometry">
       <rect>
        <x>80</x>
        <y>180</y>
        <width>51</width>
        <height>16</height>
       </rect>
      </property>
      <property name="maximum">
       <double>15.000000000000000</double>
      </property>
      <property name="singleStep">
       <double>0.010000000000000</double>
      </property>
      <property name="value">
       <double>1.000000000000000</double>
      </property>
     </widget>
     <widget class="QLabel" name="label_centerDisThrs">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>180</y>
        <width>61</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>ccDisThrs:</string>
      </property>
     </widget>
     <widget class="QDoubleSpinBox" name="doubleSpinBox_cornerDisThrs">
      <property name="geometry">
       <rect>
        <x>80</x>
        <y>200</y>
        <width>51</width>
        <height>16</height>
       </rect>
      </property>
      <property name="maximum">
       <double>10.000000000000000</double>
      </property>
      <property name="singleStep">
       <double>0.010000000000000</double>
      </property>
      <property name="value">
       <double>0.100000000000000</double>
      </property>
     </widget>
     <widget class="QLabel" name="label__cornerDisThrs">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>200</y>
        <width>71</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>cornerThrs:</string>
      </property>
     </widget>
     <widget class="QPushButton" name="pushButton_extractSpanningTree">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>80</y>
        <width>111</width>
        <height>23</height>
       </rect>
      </property>
      <property name="text">
       <string>Spanning Tree</string>
      </property>
     </widget>
     <widget class="QDoubleSpinBox" name="doubleSpinBox_mianBranchAngle">
      <property name="geometry">
       <rect>
        <x>80</x>
        <y>220</y>
        <width>51</width>
        <height>16</height>
       </rect>
      </property>
      <property name="maximum">
       <double>30.000000000000000</double>
      </property>
      <property name="singleStep">
       <double>0.010000000000000</double>
      </property>
      <property name="value">
       <double>5.000000000000000</double>
      </property>
     </widget>
     <widget class="QLabel" name="label_mianBranchAngle">
      <property name="geometry">
       <rect>
        <x>3</x>
        <y>220</y>
        <width>71</width>
        <height>20</height>
       </rect>
      </property>
      <property name="text">
       <string>mainBraAngle:</string>
      </property>
     </widget>
     <widget class="QComboBox" name="comboBox_methods">
      <property name="geometry">
       <rect>
        <x>80</x>
        <y>240</y>
        <width>51</width>
        <height>21</height>
       </rect>
      </property>
      <item>
       <property name="text">
        <string>Our</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>SA</string>
       </property>
      </item>
      <item>
       <property name="text">
        <string>GA</string>
       </property>
      </item>
     </widget>
     <widget class="QLabel" name="label_2">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>240</y>
        <width>72</width>
        <height>15</height>
       </rect>
      </property>
      <property name="text">
       <string>OptAlg.</string>
      </property>
     </widget>
    </widget>
   </widget>
  </widget>
  <action name="actionOptionDialog">
   <property name="checkable">
    <bool>false</bool>
   </property>
   <property name="checked">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>Show/hide Option Dialog</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>GlViewer</class>
   <extends>QWidget</extends>
   <header>glviewer.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
